python/pyarrow/tests/test_table.py

```diff
@@ -129,6 +130,7 @@ def ne(xarrs, yarrs):
             y = pa.chunked_array(yarrs)
         assert not x.equals(y)
         assert not y.equals(x)
+        assert x != y
```

Comment:
Test that it supports comparing with arbitrary objects, like [a string](https://github.com/apache/arrow/blob/master/python/pyarrow/tests/test_types.py#L458)